java/semantickernel-api/src/main/java/com/microsoft/semantickernel/util/Settings.java

```diff
@@ -0,0 +1,118 @@
+        private final String endpoint;
+        private final String deploymentName;
+
+        public AzureOpenAISettings(String key, String endpoint, String deploymentName) {
+            this.key = key;
+            this.endpoint = endpoint;
+            this.deploymentName = deploymentName;
+        }
+
+        public String getKey() {
+            return key;
+        }
+
+        public String getEndpoint() {
+            return endpoint;
+        }
+
+        public String getDeploymentName() {
+            return deploymentName;
+        }
+    }
+
+    private enum Property {
+        OPEN_AI_KEY("openai.key"),
+        OPEN_AI_ORGANIZATION_ID("openai.organizationid"),
+        AZURE_OPEN_AI_KEY("azureopenai.key"),
+        AZURE_OPEN_AI_ENDPOINT("azureopenai.endpoint"),
+        AZURE_OPEN_AI_DEPLOYMENT_NAME("azureopenai.deploymentname");
+
+        public final String label;
+
+        Property(String label) {
+            this.label = label;
+        }
+    }
+
+    /**
+     * Returns an instance of OpenAISettings with key and organizationId from the properties file
+     *
+     * @param path Path to the properties file
+     * @return OpenAISettings
+     */
+    public static OpenAISettings getOpenAISettingsFromFile(String path) {
+        return new OpenAISettings(
+                Settings.getSettingsValue(path, Property.OPEN_AI_KEY.label),
+                Settings.getSettingsValue(path, Property.OPEN_AI_ORGANIZATION_ID.label, ""));
+    }
+
+    /**
+     * Returns an instance of AzureOpenAISettings with key, endpoint and deploymentName from the
+     * properties file
+     *
+     * @param path Path to the properties file
+     * @return OpenAISettings
+     */
+    public static AzureOpenAISettings getAzureOpenAISettingsFromFile(String path) {
+        return new AzureOpenAISettings(
+                Settings.getSettingsValue(path, Property.AZURE_OPEN_AI_KEY.label),
+                Settings.getSettingsValue(path, Property.AZURE_OPEN_AI_ENDPOINT.label),
+                Settings.getSettingsValue(path, Property.AZURE_OPEN_AI_DEPLOYMENT_NAME.label, ""));
+    }
+
+    private static String getSettingsValue(String SettingsFile, String propertyName) {
+        return getSettingsValue(SettingsFile, propertyName, null);
+    }
+
+    private static String getSettingsValue(
+            String SettingsFile, String propertyName, String defaultValue) {
+        File Settings = new File(SettingsFile);
+        try (FileInputStream fis = new FileInputStream(Settings.getAbsolutePath())) {
+            Properties props = new Properties();
+            props.load(fis);
+            if (defaultValue == null) {
+                return props.getProperty(propertyName);
+            }
+            return props.getProperty(propertyName, defaultValue);
+        } catch (IOException e) {
+            LOGGER.error(
+                    "Unable to load config value " + propertyName + " from file " + SettingsFile,
```

Comment:
If you log the settingsFile, you leak details about the environment that this is being run in, which is a possible attack vector. Something to consider.